.github/workflows/merge-main-into-small-image.yml

```diff
@@ -0,0 +1,21 @@
+name: merge-main-into-small-image.yml
+on:
+  pull_request:
+    branches:
+      - main   
+    types:
+      - closed
+jobs:
+  merge-main-into-small-image:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    timeout-minutes: 3
+    
+    steps:
+    - uses: actions/checkout@v4
+    - name: Merge with main
+      run: |
+        git fetch
+        git checkout small-image
+        git merge main --no-ff --no-edit
```

Comment:
tested on a private personal repo. It should work unless there are merge conflicts.